pkg/skaffold/parser/config.go

```diff
@@ -169,8 +169,9 @@ func processEachConfig(config *latest_v1.SkaffoldConfig, cfgOpts configOpts, opt
 	if err := defaults.Set(config); err != nil {
 		return nil, sErrors.ConfigSetDefaultValuesErr(config.Metadata.Name, cfgOpts.file, err)
 	}
-	// convert relative file paths to absolute for all configs that are not invoked explicitly. This avoids maintaining multiple root directory information since the dependency skaffold configs would have their own root directory.
-	if cfgOpts.isDependency || opts.MakePathsAbsolute {
+	// if `opts.MakePathsAbsolute` is not set, convert relative file paths to absolute for all configs that are not invoked explicitly. This avoids maintaining multiple root directory information since the dependency skaffold configs would have their own root directory.
+	// if `opts.MakePathsAbsolute` is set, use that as condition to decide on making file paths absolute for all configs or none at all. This is used when the parsed config is marshalled out (for commands like `skaffold diagnose` or `skaffold inspect`), we want to retain the original relative paths in the output files.
```

Comment:
nit: line length too long. Please split in multiple lines so i can see it in my IDE without scrolling right.

Reply:
done, although from https://github.com/golang/go/wiki/CodeReviewComments#line-length
```
There is no rigid line length limit in Go code, but avoid uncomfortably long lines. Similarly, don't add line breaks to keep lines short when they are more readable long--for example, if they are repetitive.
```
 On most IDE you can enable `Soft Wrap` setting and it'll auto-break long lines in the viewer.